server/src/lsp-server.spec.ts

```diff
@@ -845,3 +845,38 @@ export function factory() {
         );
     }).timeout(10000);
 });
+
+describe('diagnostics (no client support)', () => {
+    before(async () => {
+        // Remove the "textDocument.publishDiagnostics" client capability.
+        const clientCapabilitesOverride = getDefaultClientCapabilities();
+        delete clientCapabilitesOverride.textDocument?.publishDiagnostics;
+        server = await createServer({
+            rootUri: null,
+            publishDiagnostics: args => diagnostics.push(args),
+            clientCapabilitesOverride
+        });
+    });
+
+    it('no diagnostics are published', async () => {
+        const doc = {
+            uri: uri('diagnosticsBar.ts'),
+            languageId: 'typescript',
+            version: 1,
+            text: `
+        export function foo(): void {
+          missing('test')
+        }
+      `
+        };
+        server.didOpenTextDocument({
+            textDocument: doc
+        });
+
+        server.requestDiagnostics();
+        await server.requestDiagnostics();
+        await new Promise(resolve => setTimeout(resolve, 200));
```

Comment:
I don't love having a fixed timeout constant here, but I did some testing locally and it looks like it's not easy to avoid, and it's an existing pattern. (might relate to a debounce internally, or the inherent slowness in an external process)

Reply:
Yes, just something that was used already and likely due to debounce.
A more explicit way would be exposing the debounce timeout and use it here.
Or just disabled debouncing for tests but then the logic would be different in tests than in production.